File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario3/id_19T2000Test.py
# id_19_Test.java
from scenario3.id_19 import sort_numbers

import pytest

class id_19_Test:
    """
    Test class of id_19.py.
    It contains ten unit test cases for the :func:`scenario3.id_19.sort_numbers`.
    """


    def test_sort_numbers_1(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('three one five') == 'one three five'
		
    def test_sort_numbers_2(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five') == 'one five nine'
		
    def test_sort_numbers_3(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five seven') == 'one five seven nine'
		
    def test_sort_numbers_4(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two') == 'one two five seven nine'
		
    def test_sort_numbers_5(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four') == 'one two four five seven nine'
		
    def test_sort_numbers_6(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six') == 'one two four five six seven nine'
		
    def test_sort_numbers_7(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six eight') == 'one two four five six seven eight nine'
		
    def test_sort_numbers_8(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six eight zero') == 'zero one two four five six seven eight nine'
		
    def test_sort_numbers_9(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six eight zero three') == 'zero one two three four five six seven eight nine'
		
    def test_sort_numbers_10(self):
        """
        Test case for the :func:`scenario3.id_19.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six eight zero three one') == 'zero one one two three four five six seven eight nine'
		
# id_20.py
from typing import List


def sort_numbers(numbers: str) -> str:
    """ Input is a space-delimited string of numberals from 'zero' to 'nine'.
    Valid choices are 'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight' and 'nine'.
    Return the string with numbers sorted from smallest to largest
    >>> sort_numbers('three one five')
    'one three five'
    """

    pass
# id_20_Test.java
from scenario3.id_20 import sort_numbers

import pytest

class id_20_Test:
    """
    Test class of id_20.py.
    It contains ten unit test cases for the :func:`scenario3.id_20.sort_numbers`.
    """
		
    def test_sort_numbers_1(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('three one five') == 'one three five'
		
    def test_sort_numbers_2(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five') == 'one five nine'
		
    def test_sort_numbers_3(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five seven') == 'one five seven nine'
		
    def test_sort_numbers_4(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two') == 'one two five seven nine'
		
    def test_sort_numbers_5(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four') == 'one two four five seven nine'
		
    def test_sort_numbers_6(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six') == 'one two four five six seven nine'
		
    def test_sort_numbers_7(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six eight') == 'one two four five six seven eight nine'
		
    def test_sort_numbers_8(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six eight zero') == 'zero one two four five six seven eight nine'
		
    def test_sort_numbers_9(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six eight zero three') == 'zero one two three four five six seven eight nine'
		
    def test_sort_numbers_10(self):
        """
        Test case for the :func:`scenario3.id_20.sort_numbers`.
        """
        assert sort_numbers('nine one five seven two four six eight zero three one') == 'zero one one two three four five six seven eight nine'
		
# id_21.py
from typing import List


def sort_numbers(numbers: str) -> str:
    """ Input is a space-delimited string of numberals from 'zero' to 'nine'.
    Valid choices are 'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight' and 'nine'.
    Return the string with numbers sorted from smallest to largest
    >>> sort_numbers('three one five')
    'one three five'
    """

    pass
# id_21_Test.java
from scenario3.id_21 import sort_numbers

import pytest

class id_21_Test:
    """
    Test class of id_21.py.
    It contains ten unit test cases for the :func:`scenario3.id_21.sort_numbers`.
    """
		
    def test_sort_numbers_1(self):
        """
        Test case for the :func:`scenario3.id_21.sort_numbers`.
        """
        assert sort_numbers('three one five') == 'one three five'
		
    def test_sort_numbers_2(self):
        """
        Test case for the :func:`scenario3.id_21.sort_numbers`.
        """
        assert sort_numbers('nine one five') == 'one five nine'
		
    def test_sort_numbers_3(self):
        """
        Test case for the :func:`scenario3.id_21

File: RQ2_Prompt_Elements/GPT3.5_Data/HumanEvalJava_output/scenario3/id_19T2000Test.java
// SortNumbers.java
package scenario3;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class SortNumbers {

    /**
     * Input is a space-delimited string of numbers from 'zero' to 'nine'.
     * Valid choices are 'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight' and 'nine'.
     * Return the string with numbers sorted from smallest to largest
     *
     * > sortNumbers("three one five")
     * "one three five"
     * > sortNumbers("")
     * ""
     * > sortNumbers("three")
     * "three"
     * > sortNumbers("three five nine")
     * "three five nine"
     * > sortNumbers("five zero four seven nine eight")
     * "zero four five seven eight nine"
     * > sortNumbers("six five four three two one zero")
     * "zero one two three four five six"
     */
    public static String sortNumbers(String numbers) {
        if (numbers == null || numbers.isEmpty()) {
            return "";
        }

        String[] numberWords = numbers.split(" ");
        Map<String, Integer> numberMap = new HashMap<>();
        numberMap.put("zero", 0);
        numberMap.put("one", 1);
        numberMap.put("two", 2);
        numberMap.put("three", 3);
        numberMap.put("four", 4);
        numberMap.put("five", 5);
        numberMap.put("six", 6);
        numberMap.put("seven", 7);
        numberMap.put("eight", 8);
        numberMap.put("nine", 9);

        Integer[] numberValues = new Integer[numberWords.length];
        for (int i = 0; i < numberWords.length; i++) {
            numberValues[i] = numberMap.get(numberWords[i]);
        }

        Arrays.sort(numberValues);

        StringBuilder result = new StringBuilder();
        for (int i = 0; i < numberValues.length; i++) {
            for (Map.Entry<String, Integer> entry : numberMap.entrySet()) {
                if (entry.getValue().equals(numberValues[i])) {
                    result.append(entry.getKey()).append(" ");
                    break;
                }
            }
        }

        return result.toString().trim();
    }
}

// SortNumbersTest.java
package scenario3;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link SortNumbers}.
* It contains ten unit test cases for the {@link SortNumbers#sortNumbers(String)} method.
*/
class SortNumbersTest {

    @Test
    void testEmptyString() {
        String result = SortNumbers.sortNumbers("");
        assertEquals("", result);
    }

    @Test
    void testSingleNumber() {
        String result = SortNumbers.sortNumbers("three");
        assertEquals("three", result);
    }

    @Test
    void testTwoNumbers() {
        String result = SortNumbers.sortNumbers("three five");
        assertEquals("three five", result);
    }

    @Test
    void testThreeNumbers() {
        String result = SortNumbers.sortNumbers("three five nine");
        assertEquals("three five nine", result);
    }

    @Test
    void testFourNumbers() {
        String result = SortNumbers.sortNumbers("five zero four seven nine eight");
        assertEquals("zero four five seven eight nine", result);
    }

    @Test
    void testSixNumbers() {
        String result = SortNumbers.sortNumbers("six five four three two one zero");
        assertEquals("zero one two three four five six", result);
    }

    @Test
    void testInvalidNumber() {
        String result = SortNumbers.sortNumbers("three five invalid");
        assertEquals("three five", result);
    }

    @Test
    void testDuplicateNumbers() {
        String result = SortNumbers.sortNumbers("three five five three");
        assertEquals("three three five five", result);
    }

    @Test
    void testNumbersWithDifferentCases() {
        String result = SortNumbers.sortNumbers("Three fIvE zErO");
        assertEquals("zErO fIvE Three", result);
    }

    @Test
    void testNumbersWithLeadingAndTrailingSpaces() {
        String result = SortNumbers.sortNumbers("  three five nine  ");
        assertEquals("three five nine", result);
    }
}